src/GitHub.Exports/Services/GitService.cs

```diff
@@ -21,6 +23,51 @@ public GitService(IRepositoryFacade repositoryFacade)
             this.repositoryFacade = repositoryFacade;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LocalRepositoryModel"/> class.
+        /// </summary>
+        /// <param name="localPath">The repository's local path.</param>
+        /// <returns>A repository model.</returns>
+        public ILocalRepositoryModel CreateLocalRepositoryModel(string localPath)
+        {
+            Guard.ArgumentNotNull(localPath, nameof(localPath));
+
+            var dir = new DirectoryInfo(localPath);
+            if (!dir.Exists)
+            {
+                throw new ArgumentException("Path does not exist", nameof(localPath));
+            }
+
+            var cloneUrl = GetUri(localPath);
+            var name = cloneUrl?.RepositoryName ?? dir.Name;
+
+            var model = new LocalRepositoryModel
+            {
+                LocalPath = localPath,
+                CloneUrl = cloneUrl,
+                Name = name,
+                Icon = Octicon.repo
+            };
+
```

Comment:
This is currently being refreshed before the branch has actually changed. :-(

Reply:
There is now a `CreateCurrentBranchModel` command instead of the `CurrentBranch` property. We'll still need to be careful about calling it too soon.

Reply:
Another comment.